src/core/featurechecklistmodel.cpp

```diff
@@ -0,0 +1,173 @@
+#include "featurechecklistmodel.h"
+#include "qgsvaluerelationfieldformatter.h"
+#include <QDebug>
+
+FeatureCheckListModel::FeatureCheckListModel()
+{
+
+}
+
+QVariant FeatureCheckListModel::data( const QModelIndex &index, int role ) const
+{
+  if ( role == CheckedRole )
+  {
+    if ( mCheckedEntries.contains( FeatureListModel::data( index, FeatureListModel::KeyFieldRole ).toString() ) )
+    {
+
+      qDebug() << "List contains : " <<  FeatureListModel::data( index, FeatureListModel::KeyFieldRole ).toString();
+    }
+    else
+    {
+
+      qDebug() << "List NOT contains : " <<  FeatureListModel::data( index, FeatureListModel::KeyFieldRole ).toString();
+    }
+
+    return mCheckedEntries.contains( FeatureListModel::data( index, FeatureListModel::KeyFieldRole ).toString() );
+  }
+  else
+    return FeatureListModel::data( index, role );
+}
+
+bool FeatureCheckListModel::setData( const QModelIndex &index, const QVariant &value, int role )
+{
+  if ( data( index, role ) == value )
+    return true;
+
+  switch ( role )
+  {
+    case CheckedRole:
+    {
+      if ( value.toBool() )
+        setChecked( index );
+      else
+        setUnchecked( index );
+      break;
+    }
+  }
+  return true;
+}
+
+QHash<int, QByteArray> FeatureCheckListModel::roleNames() const
+{
+  QHash<int, QByteArray> roles = FeatureListModel::roleNames();
+
+  roles[CheckedRole] = "checked";
+
+  return roles;
+}
+
+QVariant FeatureCheckListModel::attributeValue() const
+{
+  QVariant value;
+
+  qDebug() << "Get data from list: " <<  mCheckedEntries.join( QStringLiteral( ", " ) ).prepend( '{' ).append( '}' );
+
+  if ( mAttributeField.type() == QVariant::Map )
+  {
+    //because of json it should be stored as QVariantList
+    QVariantList vl;
+    //store as QVariantList because it's json
+    for ( const QString &s : qgis::as_const( mCheckedEntries ) )
+    {
+      // Convert to proper type
+      const QVariant::Type type { fkType() };
+      switch ( type )
+      {
+        case QVariant::Type::Int:
+          vl.push_back( s.toInt() );
+          break;
+        case QVariant::Type::LongLong:
+          vl.push_back( s.toLongLong() );
+          break;
+        default:
+          vl.push_back( s );
+          break;
+      }
+    }
+    value = vl;
+  }
+  else
+  {
+    //because of hstore it should be stored as QString
```

Comment:
I'd revert this: "if the source field only supports strings, use an hstore string"

Reply:
I would even remove the word "hstore" because it may be the "origin" of this format we use here, but it's not used for general strings instead. Like ""if the source field only supports strings, use a string"

Reply:
Notiz an mich:
      //store as QVariantList because the field type supports data structure
       //store as a formatted string because the fields supports only string